jobs/cjd_sample/src/main/java/com/google/samples/CustomAttributeSample.java

```diff
@@ -0,0 +1,203 @@
+
+/**
+ * Cloud Job Discovery Custom Attribute Samples.
+ */
+public final class CustomAttributeSample {
+
+  private static JobService jobService = JobServiceUtils.getJobService();
+
+  // [START multiple_custom_attributes_job]
+
+  /**
+   * Creates a new job with advanced custom fields (both numeric and string).
+   */
+  public static Job createJobWithNumericAndStringCustomFields(String companyName)
+      throws IOException {
+    Map<String, CustomAttribute> customAttributeMap = new HashMap<>();
+    customAttributeMap.put(
+        "skills",
+        new CustomAttribute()
+            .setStringValues(
+                new StringValues()
+                    .setValues(
+                        Arrays.asList("team management", "personal organization and risk control")))
+            .setFilterable(true));
+    customAttributeMap.put(
+        "target_bonus_rate", new CustomAttribute().setLongValue(15L).setFilterable(true));
+    customAttributeMap.put(
+        "additional_info",
+        new CustomAttribute()
+            .setStringValues(
+                new StringValues()
+                    .setValues(Arrays.asList("some additional information not searchable"))));
+    Job job =
+        new Job()
+            .setCompanyName(companyName)
+            .setRequisitionId("123456")
+            .setJobTitle("Software Engineer")
+            .setApplicationUrls(Arrays.asList("http://careers.google.com"))
+            .setDescription("Design, develop, test, deploy, maintain and improve software.")
+            .setCustomAttributes(customAttributeMap);
+    CreateJobRequest createJobRequest = new CreateJobRequest().setJob(job);
+    Job createdJob = jobService.jobs().create(createJobRequest).execute();
+    System.out.println(createdJob.getName());
+    return createdJob;
+  }
+  // [END multiple_custom_attributes_job]
+
+  /**
+   * Example filters on custom attributes.
+   */
+  public static void filtersOnCustomAttributes() throws IOException {
+
+    // [START custom_attribute_example_1]
+    String salaryCustomAttributeFilter = "(100000 <= salary) AND (salary <= 200000)";
+    JobQuery salaryJobQuery = new JobQuery().setCustomAttributeFilter(salaryCustomAttributeFilter);
+    // [END custom_attribute_example_1]
+
+    // [START custom_attribute_example_2]
+    String sponsoredJobFilter = "(NOT EMPTY(is_sponsored_job)) AND (is_sponsored_job = \"true\")";
+    JobQuery sponsorJobQuery = new JobQuery().setCustomAttributeFilter(sponsoredJobFilter);
+    // [END custom_attribute_example_2]
+
+    // [START custom_attribute_example_3]
+    String langudageWithAccessControlFlag =
+        "      ( "
+            + "    (text = \"lang1\") "
+            + "    AND (text2 = \"lang2\") "
+            + "    AND (text3 = \"true\") "
+            + "    AND (ACF1 = \"languages\") "
+            + "    AND (ACF2 = \"language list\")"
+            + ") "
+            + "OR "
+            + "(ACF3 = \"true\" OR ACF3 = \"false\")";
```

Comment:
What does this do?

Reply:
You might refer https://cloud.google.com/job-discovery/docs/reference/rest/v2/JobQuery#FIELDS.custom_attribute_filter